python/example/two_cell_gap_junctions.py

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/env python3
+
+import arbor
+import argparse
+import numpy as np
+
+import pandas
+import seaborn  # You may have to pip install these.
+import matplotlib.pyplot as plt
+
+
+class TwoCellsWithGapJunction(arbor.recipe):
+    def __init__(self, probes,
+                 Vms, length, radius, cm, rL, g, gj_g,
+                 cv_policy_max_extent):
+        """
+        probes -- list of probes
+
+        Vm -- membrane leak potential
+        length -- length of cable in μm
+        radius -- radius of cable in μm
+        cm -- membrane capacitance in F/m^2
+        rL -- axial resistivity in Ω·cm
+        g -- membrane conductivity in S/cm^2
+        gj_g -- gap junction conductivity in μS
+
+        cv_policy_max_extent -- maximum extent of control volume in μm
+        """
+
+        # (4.1) The base C++ class constructor must be called first, to ensure that
+        # all memory in the C++ class is initialized correctly.
+        arbor.recipe.__init__(self)
+
+        self.the_probes = probes
+
+        self.Vms = Vms
+        self.length = length
+        self.radius = radius
+        self.cm = cm
+        self.rL = rL
+        self.g = g
+        self.gj_g = gj_g
+
+        self.cv_policy_max_extent = cv_policy_max_extent
+
+        self.the_props = arbor.neuron_cable_properties()
+        self.the_cat = arbor.default_catalogue()
+        self.the_props.register(self.the_cat)
+
+    def num_cells(self):
+        return 2
+
+    def num_sources(self, gid):
+        assert gid in [0, 1]
+        return 0
+
+    def cell_kind(self, gid):
+        assert gid in [0, 1]
+        return arbor.cell_kind.cable
+
+    def cell_description(self, gid):
+        """A high level description of the cell with global identifier gid.
+
+        For example the morphology, synapses and ion channels required
+        to build a multi-compartment neuron.
+        """
+        assert gid in [0, 1]
+
+        tree = arbor.segment_tree()
+
+        tree.append(arbor.mnpos,
+                    arbor.mpoint(0, 0, 0, self.radius),
+                    arbor.mpoint(self.length, 0, 0, self.radius),
+                    tag=1)
+
+        labels = arbor.label_dict({'cell' : '(tag 1)', 'gj_site': '(location 0 0.5)'})
+
+        decor = arbor.decor()
+        decor.set_property(Vm=self.Vms[gid])
+        decor.set_property(cm=self.cm)
+        decor.set_property(rL=self.rL)
+
+        junction_mech = arbor.junction('gj', {"g" : 1})
+        decor.place('"gj_site"', junction_mech, 'gj')
```

Comment:
Can we add a comment here saying that this adds a gap junction mechanism at the "gj_site" location and labels the mechanism 'gj'?
Also, maybe we can rename the label to 'gj_label' to avoid confusion with the mechanism name?